spec/cohortManager.spec.js

```diff
@@ -0,0 +1,441 @@
+const CohortManager = require('../src/cohortManager.js')
+
+describe('cohortManager', () => {
+  let cohortManager
+
+  beforeEach(() => {
+    cohortManager = new CohortManager()
+  })
+
+  // Test 1
+  it('creates a cohort', () => {
+    // Set up
+    const expected = [
+      {
+        cohortName: 'Cohort 9',
+        students: []
+      },
+      {
+        cohortName: 'Cohort 10',
+        students: []
+      }
+    ]
+    // Execute
+    cohortManager.create('Cohort 9')
+    const result = cohortManager.create('Cohort 10')
+    // Verify
+    expect(result).toEqual(expected)
+  })
+
+  // Test 2
+  it('returns the cohort data, if it exists', () => {
+    // Set up
+    const expected = {
+      cohortName: 'Cohort 8',
+      students: []
+    }
+
+    // Execute
+    cohortManager.create('Cohort 7')
+    cohortManager.create('Cohort 8')
+    cohortManager.create('Cohort 9')
+    cohortManager.create('Cohort 10')
+    const result = cohortManager.searchCohort('Cohort 8')
+
+    // Verify
+    expect(result).toEqual(expected)
+  })
+
+  // Test 3
+  it('Throws error message if the cohort being serached does not exist', () => {
+    // Given
+    const cohort = 'Cohort 12'
+
+    // Execute
+    cohortManager.create('Cohort 7')
+    cohortManager.create('Cohort 8')
+    cohortManager.create('Cohort 9')
+    cohortManager.create('Cohort 10')
+
+    // verify
+    expect(() => cohortManager.searchCohort(cohort)).toThrowError(
+      "The cohort doesn't exist"
+    )
+  })
+
+  // Test 4
```

Comment:
does this test take into account adding a student to a specific cohort?

Reply:
The way I have structured my local datastore is:

```
this.cohortData = [
      {
        cohortName: 'Cohort 9',
        students: [
          {
            firstName: 'Kyle',
            lastName: 'Bridgewater',
            githubUsername: 'kyleUnderwater',
            email: 'kylebridge@yahoo.com',
            id: 111
          },
          {
            firstName: 'Billy',
            lastName: 'Sanders',
            githubUsername: 'billysanders101',
            email: 'billysanders101@gmail.com',
            id: 112
          }
        ]
      },
      {
        cohortName: 'Cohort 10',
        students: [
          {
            firstName: 'Alexandra',
            lastName: "O'neil",
            githubUsername: 'zandOneil4',
            email: 'AlexOneil@gmail.com',
            id: 211
          }
        ]
      }
    ]
```

I have an array as the value of the students key, and that array is populated with objects containing the student data. There is also the cohort that they belong to shown in the parent object:

```
{
        cohortName: 'Cohort 9',
        students: [
          {
            firstName: 'Kyle',
            lastName: 'Bridgewater',
            githubUsername: 'kyleUnderwater',
            email: 'kylebridge@yahoo.com',
            id: 111
          },
          {
            firstName: 'Billy',
            lastName: 'Sanders',
            githubUsername: 'billysanders101',
            email: 'billysanders101@gmail.com',
            id: 112
          }
        ]
      }
```